napi/oxlint2/src-js/index.js

```diff
@@ -51,41 +51,39 @@ async function loadPluginImpl(path) {
 
   // TODO: Use a validation library to assert the shape of the plugin, and of rules
   const pluginName = plugin.meta.name;
-  let ruleId = registeredRules.length;
+  const offset = registeredRules.length;
   const ruleNames = [];
-  const ret = { name: pluginName, offset: ruleId, ruleNames };
 
   for (const [ruleName, rule] of Object.entries(plugin.rules)) {
     ruleNames.push(ruleName);
-    registeredRules.push({ rule, context: new Context(ruleId, `${pluginName}/${ruleName}`) });
-    ruleId++;
+    registeredRules.push({ rule, context: new Context(`${pluginName}/${ruleName}`) });
```

Comment:
unrelated, but just noticed, should we re-use the string for the `pluginName/ruleName` rather than creating it? might need benching

Reply:
How do you mean? This is when rule is registered, which only happens once. So I think each string is only created once.

We could do this:

```js
const pluginNameWithSlash = `${pluginName}/`;
for (const [ruleName, rule] of Object.entries(plugin.rules)) {
  ruleNames.push(ruleName);
  registeredRules.push({ rule, context: new Context(`${pluginNameWithSlash}${ruleName}`) });
}
```

But I'm not sure if that'd be a perf improvement or not. If it is, I doubt the difference would be measurable.

Reply:
probably won't have a perf diff, but i more meant changing `plugin.rules shape to be like
```
type Rules = { [ruleName: string]: { fullName: `${pluginName}/${ruleNme}`, rule: any } }
```

so we only construct `fullName` once. not important to optimize right now though.

Currently we construct `${pluginNameWithSlash}${ruleName}`` num files * num rules times, but we could just change it to construct a max of `num rules` times.